src/simulator/short_term_object.cpp

```diff
@@ -0,0 +1,71 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    short_term_object.cpp
+  \brief   C++ Interface: Short term object class
+  \author  Yifeng Zhu, (C) 2020
+  \email   yifeng.zhu@utexas.edu
+*/
+//========================================================================
+
+#include "short_term_object.h"
+
+ShortTermObject::ShortTermObject(){
+
+  // x, y, angle
+  pose_ = Eigen::Vector3f(0., 0., 0.);
+
+  // Example of a simple shape
+  double r = 0.5;
+  double eps = 0.001;
+  template_lines_.push_back(geometry::line2f(Eigen::Vector2f(r - eps, r), Eigen::Vector2f(-r + eps, r)));
+  template_lines_.push_back(geometry::line2f(Eigen::Vector2f(-r, r - eps), Eigen::Vector2f(-r, -r + eps)));
+  template_lines_.push_back(geometry::line2f(Eigen::Vector2f(-r + eps, -r), Eigen::Vector2f(r - eps, -r)));
+  template_lines_.push_back(geometry::line2f(Eigen::Vector2f(r, -r + eps), Eigen::Vector2f(r, r - eps)));
+
+  pose_lines_ = template_lines_;
+}
+
+ShortTermObject::ShortTermObject(std::string config_file){
+  pose_ = Eigen::Vector3f(0., 0., 0.);
+  // TODO replace manually defined in the code with loading initial locations
```

Comment:
When this PR is accepted you should add issues for remaining todos such as this, and for using configs for additional shapes.